pkg/controller/unifiedpushserver/unifiedpushserver.go

```diff
@@ -89,7 +89,85 @@ func newOauthProxyImageStream(cr *pushv1alpha1.UnifiedPushServer) (*imagev1.Imag
 	}, nil
 }
 
-func newUnifiedPushServerDeploymentConfig(cr *pushv1alpha1.UnifiedPushServer) (*openshiftappsv1.DeploymentConfig, error) {
+func buildEnv(cr *pushv1alpha1.UnifiedPushServer, addressSpaceURL string) []corev1.EnvVar {
+	var env = []corev1.EnvVar{
+		{
+			Name: "POSTGRES_SERVICE_HOST",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "POSTGRES_HOST",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: fmt.Sprintf("%s-postgresql", cr.Name),
+					},
+				},
+			},
+		},
+		{
+			Name:  "POSTGRES_SERVICE_PORT",
+			Value: "5432",
+		},
+		{
+			Name: "POSTGRES_USER",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "POSTGRES_USERNAME",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: fmt.Sprintf("%s-postgresql", cr.Name),
+					},
+				},
+			},
+		},
+		{
+			Name: "POSTGRES_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "POSTGRES_PASSWORD",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: fmt.Sprintf("%s-postgresql", cr.Name),
+					},
+				},
+			},
+		},
+		{
+			Name: "POSTGRES_DATABASE",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "POSTGRES_DATABASE",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: fmt.Sprintf("%s-postgresql", cr.Name),
+					},
+				},
+			},
+		},
+	}
+
+	if cr.Spec.UseMessageBroker {
+		env = append(env,
+			corev1.EnvVar{
```

Comment:
I see that now there's a Secret (for the password) and a ConfigMap (for the host). Are those just being created so that they can be referenced in these EnvVars in the UPS deploymentconfig? If so, that may or may not be worthwhile. :thinking: What I was wondering in the question above was more like "incidentally, if they happen to be in a Secret or ConfigMap, we could wire them up from there".